### DISCUSSION TITLE
How do I enable blink.cmp lsp capabilities while using lsp-zero?

### DISCUSSION THREAD

**Discussion by allengueco** [VonHeikemen/lsp-zero.nvim#419] [Q&A] [answered]
When using [blink.cmp](https://github.com/Saghen/blink.cmp) instead of nvim-cmp, they suggest doing something like this:
```lua
{
  'neovim/nvim-lspconfig',
  dependencies = { 'saghen/blink.cmp' },
  config = function(_, opts)
    local lspconfig = require('lspconfig')
    for server, config in pairs(opts.servers) do
      config.capabilities = require('blink.cmp').get_lsp_capabilities(config.capabilities)
      lspconfig[server].setup(config)
    end
  end
}
```
How would I do this in lsp-zero?

**Comment by VonHeikemen** [comment, marked as answer]
> How would I do this in lsp-zero?

In this case you just do it. Since `version 4` lsp-zero doesn't have any effect on the plugins you use. You don't integrate plugins with lsp-zero anymore.

---

To use `blink.cmp` with `lspconfig` there are a couple of options:

Option 1: add capabilities manually.

Get the `capabilities` from `blink.cmp` and put them in the setup of each language server.

```lua
local lsp_capabilities = require('blink.cmp').get_lsp_capabilities()

--- these are just example language servers
--- replace them with the ones you actually have installed
require('lspconfig').gleam.setup({
  capabilities = lsp_capabilities,
})
require('lspconfig').gopls.setup({
  capabilities = lsp_capabilities,
})
```

Option 2: add the capabilities to `lspconfig` defaults.

```lua
-- Add blink.cmp capabilities settings to lspconfig
-- This should be executed before you configure any language server
local lspconfig_defaults = require('lspconfig').util.default_config
lspconfig_defaults.capabilities = vim.tbl_deep_extend(
  'force',
  lspconfig_defaults.capabilities,
  require('blink.cmp').get_lsp_capabilities()
)
```

**Reply by allengueco** [reply]
Perfect! I guess I was a little confused since `.get_lsp_capabilities()` from their example took the server's capabilities as an argument.

Just out of curiosity, could I also have used `lsp_zero.client_config()` for this?

**Reply by VonHeikemen** [reply]
> Just out of curiosity, could I also have used lsp_zero.client_config() for this?

Yes. That also works.
